src/e3/pytest.py

```diff
@@ -0,0 +1,235 @@
+        for regex in DEFAULT_EXCLUDE_LIST:
+            cov.exclude(regex)
+
+        # Exclude all <os>-only line (unless that's for this OS)
+        for regex in (
+            "%s-only" % o
+            for o in ("darwin", "linux", "solaris", "windows", "bsd", "aix")
+            if o != os_name
+        ):
+            cov.exclude(regex)
+
+        # Handling of Windows
+        if os_name == "windows":
+            cov.exclude('if sys.platform != "win32":')
+        else:
+            cov.exclude('if sys.platform == "win32":')
+            cov.exclude("unix: no cover")
+
+        # Exclude no cover for this OS
+        cov.exclude(f"{os_name}: no cover")
+
+        cov.load()
+
+        # Read configuration files in <root>/tests/coverage to fix the list
+        # of files to omit for OS specific list
+        # We're relying on the default coverage configuration for the
+        # platform-agnostic list
+        omit_files: list[str] = cov.get_option("run:omit") or []  # type: ignore
+        coverage_conf_dir = session.config.rootpath / "tests" / "coverage"
+        conf_file = coverage_conf_dir / f"omit-files-{os_name}"
+        if conf_file.exists():
+            with conf_file.open() as f:
+                for line in f:
+                    omit_files.append(line.rstrip())
+
+        # cov.html_report(directory=str(session.config.rootpath ), omit=omit_files)
+        cov.html_report(omit=omit_files)
+        cov.report(omit=omit_files, precision=3)
+        cov.xml_report()
+
+
+def fix_coverage_paths(origin_dir: str, new_dir: str, cov_db: str) -> None:
+    """Fix coverage paths.
+
+    :param origin_dir: path to the package directory, e.g.
+        .tox/py311-cov-xdist/lib/python3.11/site-packages/e3
+    :param new_dir: path to the dir that should be visible instead of origin_dir
+        e.g. src/
+    :param cov_db: path to the .coverage data base
+    """
+    paths = PathAliases()
+    paths.add(origin_dir, new_dir)
+
+    old_cov_file = NamedTemporaryFile(dir=os.path.dirname(cov_db))
+    old_cov_file.close()
+    try:
+        mv(cov_db, old_cov_file.name)
+
+        old_coverage_data = CoverageData(old_cov_file.name)
+        old_coverage_data.read()
+        new_coverage_data = CoverageData(cov_db)
+        new_coverage_data.update(old_coverage_data, aliases=paths)
+        new_coverage_data.write()
+    finally:
+        os.unlink(old_cov_file.name)
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item: pytest.Item, call: pytest.CallInfo[None]) -> None:  # type: ignore
+    """Generate results file.
+
+    When the variable results_dir is set to an existing directory, the testsuite
+    will generate results file in "anod" format.
+    """
+    global test_errors
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    rep = outcome.get_result()
+    results_dir = os.environ.get("RESULTS_DIR")
+
+    if not results_dir or not os.path.isdir(results_dir):
```

Comment:
If the directory does not exist, I think we could either let the user know it, or create it.